marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -44,6 +44,27 @@ import Welcome.Lenses (_walletLibrary)
 import Welcome.State (handleAction, dummyState, mkInitialState) as Welcome
 import Welcome.Types (Action, State) as Welcome
 
+{-
+The Marlowe Run app consists of six main workflows:
+
+1. Generate a demo wallet (this will become redundant when we integrate with real wallets).
+2. Connect a wallet (this will change when we integrate with real wallets).
+3. Disconnect a wallet.
+4. Start a contract.
+5. Move a contract forward.
+6. Redeem payments (this is triggered automatically - but may need to be manual when we
+   integrate with real wallets).
+
+There are two main application states: the `Welcome` state (for workflows 1 and 2), and the
+`Dashboard` state (for workflows 3-6). Initially we are in the `Welcome` state. Connecting a wallet
+(workflow 2) moves you into the `Dashboard` state; disconnecting a wallet (workflow 4) moves you
+back into the `Welcome` state.
+
+Because of the synchronous nature of the app, with messages passing between the browser and the PAB
```

Comment:
asynchronous... and BTW, part of the requestId mechanism that I'm putting in place could result in us "blocking" an  action handler (without blocking other actions) until we have the response from the server.